src/progressbar/progressbar.ts

```diff
@@ -0,0 +1,36 @@
+import {Component, Input, ElementRef} from 'angular2/angular2';
+
+@Component({
+  selector: 'ngb-progressbar',
+  template: `
+    <progress class="progress {{extraClasses}}  {{type && 'progress-' + type}}" [value]="value" [max]="max">
+      <div class="progress">
+        <span class="progress-bar" [style.width.%]="calculatePorcentage()"><ng-content></ng-content></span>
+      </div>
+    </progress>
+  `
+})
+export class NgbProgressbar {
+  _value: number;
+  min = 0;
+  extraClasses: string;
+
+  @Input()
+  set value(newValue: number) {
+    this._value = newValue;
+    this.calculatePorcentage();
+  }
+
+  get value() { return this._value; }
+
+  @Input() type: string;
+
+  @Input() max = 100;
+
+  constructor(private element: ElementRef) {
```

Comment:
We can do everything without injecting `ElementRef`:
* you can get classes set on an element by doing a setter for `class`, see: https://github.com/angular/angular/blob/master/modules/angular2/src/core/directives/ng_class.ts
* this will also make sure that the class is not propagated to the host element so you won't need to clear it by hand

In short: I don't think we need / should be injecting `ElementRef` here